model/keyvalue.go

```diff
@@ -19,45 +19,42 @@ import (
 	"encoding/hex"
 	"fmt"
 	"io"
-	"math"
 	"sort"
 	"strconv"
+	"strings"
 )
 
-// ValueType describes the type of value contained in a KeyValue struct
-type ValueType int
-
 const (
 	// StringType indicates the value is a unicode string
-	StringType ValueType = iota
+	StringType = ValueType_STRING
 	// BoolType indicates the value is a Boolean encoded as int64 number 0 or 1
-	BoolType
+	BoolType = ValueType_BOOL
 	// Int64Type indicates the value is an int64 number
-	Int64Type
+	Int64Type = ValueType_INT64
 	// Float64Type indicates the value is a float64 number stored as int64
-	Float64Type
+	Float64Type = ValueType_FLOAT64
 	// BinaryType indicates the value is binary blob stored as a byte array
-	BinaryType
+	BinaryType = ValueType_BINARY
 
-	stringTypeStr  = "string"
-	boolTypeStr    = "bool"
-	int64TypeStr   = "int64"
-	float64TypeStr = "float64"
-	binaryTypeStr  = "binary"
+	// stringTypeStr  = "string"
+	// boolTypeStr    = "bool"
+	// int64TypeStr   = "int64"
+	// float64TypeStr = "float64"
+	// binaryTypeStr  = "binary"
 )
 
 // KeyValue describes a tag or a log field that consists of a key and a typed value.
 // Before accessing a value, the caller must check the type. Boolean and numeric
 // values should be accessed via accessor methods Bool(), Int64(), and Float64().
 //
 // This struct is designed to minimize heap allocations.
-type KeyValue struct {
-	Key   string    `json:"key"`
-	VType ValueType `json:"vType"`
-	VStr  string    `json:"vStr,omitempty"`
-	VNum  int64     `json:"vNum,omitempty"`
-	VBlob []byte    `json:"vBlob,omitempty"`
-}
+// type KeyValue struct {
+// 	Key   string    `json:"key"`
+// 	VType ValueType `json:"vType"`
+// 	VStr  string    `json:"vStr,omitempty"`
+// 	VNum  int64     `json:"vNum,omitempty"`
```

Comment:
the KeyValue struct is now less memory efficient because `vNum` or split into `vBool/vInt64/vFloat64`